runtime/executor/src/test/java/org/apache/nemo/runtime/executor/task/TaskExecutorTest.java

```diff
@@ -391,21 +391,21 @@ private StageEdge mockBroadcastVariableStageEdgeTo(final IRVertex srcVertex,
    * Represents the answer return an inter-stage {@link InputReader},
    * which will have multiple iterable according to the source parallelism.
    */
-  private class ParentTaskReaderAnswer implements Answer<InputReader> {
+  private class ParentTaskReaderAnswer implements Answer<BlockInputReader> {
     @Override
-    public InputReader answer(final InvocationOnMock invocationOnMock) throws Throwable {
+    public BlockInputReader answer(final InvocationOnMock invocationOnMock) throws Throwable {
       final List<CompletableFuture<DataUtil.IteratorWithNumBytes>> inputFutures = new ArrayList<>(SOURCE_PARALLELISM);
       final int elementsPerSource = DATA_SIZE / SOURCE_PARALLELISM;
       for (int i = 0; i < SOURCE_PARALLELISM; i++) {
         inputFutures.add(CompletableFuture.completedFuture(
             DataUtil.IteratorWithNumBytes.of(elements.subList(i * elementsPerSource, (i + 1) * elementsPerSource)
                 .iterator())));
       }
-      final InputReader inputReader = mock(InputReader.class);
+      final BlockInputReader inputReader = mock(BlockInputReader.class);
```

Comment:
Do we have test codes for PipeInputReader and PipeOutputWriter?

Reply:
Thanks for the catch. I've slipped this one. I changed this for while debugging, but this really should not change as we should mock the InputReader/OutputReader interfaces and not their implementations.